api/test-context/src/test_context.rs

```diff
@@ -116,17 +100,7 @@ pub fn new_test_context(test_name: String, api_version: &str) -> TestContext {
     let (validator_identity, _, _) = validators[0].get_key_objects(None).unwrap();
     let validator_owner = validator_identity.account_address.unwrap();
 
-    let (db, db_rw) = DbReaderWriter::wrap(
-        AptosDB::open(
-            &tmp_dir,
-            false,                       /* readonly */
-            NO_OP_STORAGE_PRUNER_CONFIG, /* pruner */
-            RocksdbConfigs::default(),
-            true, /* indexer */
-            TARGET_SNAPSHOT_SIZE,
-        )
-        .unwrap(),
-    );
+    let (db, db_rw) = DbReaderWriter::wrap(AptosDB::new_for_test_with_indexer(&tmp_dir));
```

Comment:
we should not default this for all tests- we'll want to make sure it's disabled for non-indexer use cases, so that they don't accidentally end up relying on a DB API that they won't actually have

Reply:
Alright I'll add a flag.